packages/synapse-react-client/src/components/AccessRequirementAclEditor/AccessRequirementAclEditor.tsx

```diff
@@ -0,0 +1,271 @@
+import { Alert, Box, Skeleton, Stack, Typography } from '@mui/material'
+import {
+  ACCESS_TYPE,
+  AccessControlList,
+  ResourceAccess,
+} from '@sage-bionetworks/synapse-types'
+import { isEqual } from 'lodash-es'
+import React, { useEffect, useImperativeHandle, useState } from 'react'
+import {
+  useCreateAccessRequirementACL,
+  useDeleteAccessRequirementACL,
+  useGetAccessRequirementACL,
+  useUpdateAccessRequirementACL,
+} from '../../synapse-queries'
+import UserSearchBoxV2 from '../UserSearchBox/UserSearchBoxV2'
+import { ResourceAccessItem } from './ResourceAccessItem'
+
+const textSx = {
+  variant: 'body1',
+  lineHeight: '20px',
+  color: 'grey.800',
+}
+
+export const EMPTY_RESOURCE_ACCESS_LIST_TEXT =
+  'Only ACT has permissions on this AR.'
+export const REVIEWER_ALREADY_ADDED_ERROR_MESSAGE =
+  'User or team already has permissions on this AR.'
+
+export type AccessRequirementAclEditorHandle = {
+  /* Allow the parent component to trigger saving the ACL, so this may be embedded in an arbitrary modal. */
+  save: () => void
+}
+
+export type AccessRequirementAclEditorProps = {
+  accessRequirementId: string
+  /* Called when ACL has been saved or an error has been returned */
+  onSaveComplete: (saveSuccessful: boolean) => void
+}
+
+export const AccessRequirementAclEditor = React.forwardRef(
+  function AccessRequirementAclEditor(
+    props: AccessRequirementAclEditorProps,
+    ref: React.ForwardedRef<AccessRequirementAclEditorHandle>,
+  ) {
+    const { accessRequirementId, onSaveComplete } = props
+    const [error, setError] = useState<string | null>(null)
+    const [resourceAccessList, setResourceAccessList] = useState<
+      ResourceAccess[]
+    >([])
+
+    const { data: originalAcl, isLoading: isLoadingOriginalAcl } =
+      useGetAccessRequirementACL(accessRequirementId, {
+        // Infinite staleTime ensures this won't get re-fetched unless explicitly invalidated by the mutation
+        staleTime: Infinity,
+      })
+
+    // set resourceAccessList when the fetched acl changes
+    useEffect(() => {
+      if (originalAcl) {
+        setResourceAccessList(originalAcl.resourceAccess)
+      }
+    }, [originalAcl])
+
+    const { mutate: deleteAcl } = useDeleteAccessRequirementACL({
+      onSuccess: () => {
+        setError(null)
+        onSaveComplete(true)
+      },
+      onError: error => {
+        setError(error.reason)
+        onSaveComplete(false)
+      },
```

Comment:
Suggest creating local fns `onMutateSuccess` and `onMutateError` and re-using them in each of these mutation hook calls